pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionHandler.java

```diff
@@ -104,11 +108,15 @@ protected void reconnectLater(Throwable exception) {
         state.setState(State.Connecting);
         state.client.timer().newTimeout(timeout -> {
             log.info("[{}] [{}] Reconnecting after connection was closed", state.topic, state.getHandlerName());
-            ++epoch;
+            incrementEpoch();
             grabCnx();
         }, delayMs, TimeUnit.MILLISECONDS);
     }
 
+    protected long incrementEpoch() {
```

Comment:
Maybe we can keep this `private`, unless we need to expose.

Reply:
that's true. I exposed it as `protected` since the `reconnectLater` method is `protected` and I was following a similar style.